subsys/net/lib/lwm2m/lwm2m_engine.c

```diff
@@ -141,45 +106,62 @@ static struct lwm2m_ctx *sock_ctx[MAX_POLL_FD];
 static struct pollfd sock_fds[MAX_POLL_FD];
 static int sock_nfds;
 
+/* Resource wrappers */
+struct lwm2m_ctx **lwm2m_sock_ctx(void)
+{
+	return sock_ctx;
+}
+
+int lwm2m_sock_nfds(void)
```

Comment:
Please add the declaration to the header file. Make the same for all the other functions which are used from _lwm2m_observation.h_.

Reply:
I was wondering same that own header file would be useful in this change.